File: JScript/clsPfJointLoad.js
//
// Copyright (c)2014 S C Harrison
// Refer to License.txt for terms and conditions of use.
//



//Define Class
function clsPfJointLoad()
{
 key = 0;

 jt = 0;
 fx = 0;          //.. horizontal load @ a joint ..
 fy = 0;          //.. vertical   load @ a joint ..
 mz = 0;          //.. moment applied  @ a joint ..
}


clsPfJointLoad.prototype.initialise = function()
{
  this.key = 0;
  this.jt = 0;
  this.fx = 0;
  this.fy = 0;
  this.mz = 0;
}

clsPfJointLoad.prototype.setValues = function(LoadKey, Node, ForceX, ForceY, Moment)
{
  this.key = LoadKey;
  this.jt = Node;
  this.fx = ForceX;
  this.fy = ForceY;
  this.mz = Moment;
}

clsPfJointLoad.prototype.sprint = function()
{
  var s;
  
  s = "";
  s = s + StrLPad(this.key.toString(), 8);
  s = s + StrLPad(this.jt.toString(), 6);
  s = s + StrLPad(this.fx.toString(), 15);
  s = s + StrLPad(this.fy.toString(), 15);
  s = s + StrLPad(this.mz.toString(), 15);
               
  return s
}

clsPfJointLoad.prototype.cprint = function()
{
  WScript.Echo(this.sprint());
}

clsPfJointLoad.prototype.fprint = function(fp)
{
  fp.WriteLine(this.sprint());
}

clsPfJointLoad.prototype.fgetData = function(fp)
{
  var s;
  var n;
  var dataflds;
  var i,n;

  WScript.Echo( "fgetData ...");
  
   s = fp.ReadLine();
   WScript.Echo( s);
   this.sgetData(s);

   WScript.Echo( "... fgetData");
}

clsPfJointLoad.prototype.sgetData = function(s)
{
  var n;
  var dataflds;
  var i,n;

  WScript.Echo( "sgetData ...");
  
   WScript.Echo( s);

   s = s.replace(/^\s+|\s+$/gm,''); //trim trailing spaces
   dataflds = s.match(/-?\d+(?:[,.]\d+)?/gi);
   
   n = dataflds.length;
   for (i=0;i<n;i++)
   {
    WScript.Echo( i + "<" + dataflds[i] + ">" );
   }
   
    this.key = parseInt(dataflds[0]);
    this.jt = parseInt(dataflds[1]);
    this.fx = parseFloat(dataflds[2]);
    this.fy = parseFloat(dataflds[3]);
    this.mz = parseFloat(dataflds[4]);


   WScript.Echo( "... sgetData");
}

File: VisualStudio/CSharp/w7cpframe/clsPfJointLoad.cs
//
// Copyright (c)2016 S C Harrison
// Refer to License.txt for terms and conditions of use.
//
using System;
using System.Collections.Generic;
using System.Text;
using System.IO;
using System.Text.RegularExpressions;


namespace w7cpframe
{
    class clsPfJointLoad
    {


        public int key = 0;

        public int jt = 0;
        public double fx = 0;          //.. horizontal load @ a joint ..
        public double fy = 0;          //.. vertical   load @ a joint ..
        public double mz = 0;          //.. moment applied  @ a joint ..



        public void initialise()
        {
            key = 0;
            jt = 0;
            fx = 0;
            fy = 0;
            mz = 0;
        }

        public void setValues(int LoadKey, int Node, double ForceX, double ForceY, double Moment)
        {
            key = LoadKey;
            jt = Node;
            fx = ForceX;
            fy = ForceY;
            mz = Moment;
        }

        public string sprint()
        {
            string s;

            s = "";
            s = s + key.ToString().PadLeft(8);
            s = s + jt.ToString().PadLeft(6);
            s = s + fx.ToString().PadLeft(15);
            s = s + fy.ToString().PadLeft(15);
            s = s + mz.ToString().PadLeft(15);

            return s;
        }

        public void cprint()
        {
            System.Console.WriteLine(sprint());
        }

        public void fprint(StreamWriter fp)
        {
            fp.WriteLine(sprint());
        }

        public void fgetData(StreamReader fp)
        {
            string s;

            System.Console.WriteLine("fgetData ...");

            s = fp.ReadLine();
            System.Console.WriteLine(s);
            sgetData(s);

            System.Console.WriteLine("... fgetData");
        }

        public void sgetData(string s)
        {
            string[] dataflds = new string[10]; //(0 To 9);
            char[] delimters = new char[] { ' ', '\t' };
            int i, n;

            System.Console.WriteLine("sgetData ...");

            System.Console.WriteLine(s);

            string regResult;
            string pattern = @"^\s+|\s+$"; //trim trailing spaces
            regResult = Regex.Replace(s, pattern, "");

            dataflds.Initialize();
            string pattern2 = @"-?\d+(?:[,.]\d+)?";
            i = 0;
            foreach (Match match in Regex.Matches(regResult, pattern2))
            {
                if (match.Value != "")
                {
                    dataflds[i] = match.Value;
                    System.Console.WriteLine(match.Value);
                    i++;
                }
            }

            n = dataflds.Length;
            for (i = 0; i < n; i++)
            {
                System.Console.WriteLine(i + "<" + dataflds[i] + ">");
            }

            key = int.Parse(dataflds[0]);
            jt = int.Parse(dataflds[1]);
            fx = double.Parse(dataflds[2]);
            fy = double.Parse(dataflds[3]);
            mz = double.Parse(dataflds[4]);


            System.Console.WriteLine("... sgetData");
        }




    } //class
}  //name space
